crates/orderbook/src/solvable_orders.rs

```diff
@@ -327,41 +334,64 @@ impl Maintaining for SolvableOrdersCache {
 async fn get_orders_with_native_prices(
     mut orders: Vec<Order>,
     native_price_estimator: &dyn NativePriceEstimating,
+    deadline: Instant,
 ) -> (Vec<Order>, BTreeMap<H160, U256>) {
     let traded_tokens = orders
         .iter()
         .flat_map(|order| [order.creation.sell_token, order.creation.buy_token])
         .collect::<HashSet<_>>()
         .into_iter()
         .collect::<Vec<_>>();
-    let prices = native_vec_estimates(native_price_estimator, &traded_tokens)
-        .await
-        .into_iter()
-        .zip(traded_tokens)
-        .filter_map(|(price, token)| match price {
-            Ok(price) => Some((token, to_normalized_price(price)?)),
-            Err(err) => {
-                tracing::warn!(?token, ?err, "error estimating native token price");
-                None
-            }
-        })
-        .collect::<BTreeMap<_, _>>();
+    let mut prices = BTreeMap::new();
```

Comment:
Might make sense to have this be a `HashMap` for faster access. We return the `used_prices` which needs to be a `BTreeMap`.
```suggestion
    let mut prices = HashMap::new();
```